distributed/shuffle/shuffle_extension.py

```diff
@@ -57,22 +65,25 @@ def __init__(
         self.schema = schema
         self.worker = worker
         self.output_workers = output_workers
-        self.executor = executor
+        self.executor = ThreadPoolExecutor(worker.state.nthreads)
```

Comment:
This slipped in. I wanted to clean up some abstraction leackage between Shuffle and WorkerExtension. Will follow up with a couple of other changes. This is small enough that I didn't cherry pick it